compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -507,12 +507,12 @@ object TypeOps:
    *  does not update `ctx.nestingLevel` when entering a block so I'm leaving
    *  this as Future Work™.
    */
-  def avoid(tp: Type, symsToAvoid: => List[Symbol])(using Context): Type = {
+  def avoid(tp: Type, symsToAvoid: => List[Symbol], avoidStatic: Boolean = false)(using Context): Type = {
     val widenMap = new AvoidMap {
       @threadUnsafe lazy val forbidden = symsToAvoid.toSet
       def toAvoid(tp: NamedType) =
         val sym = tp.symbol
-        !sym.isStatic && forbidden.contains(sym)
+        (!sym.isStatic || avoidStatic) && forbidden.contains(sym)
```

Comment:
What would go wrong if we just dropped the first condition? Only ask `forbidden.contains(sym)`?